modules/web/server/src/main/scala/observe/web/server/http4s/vars.scala

```diff
@@ -4,9 +4,30 @@
 package observe.web.server.http4s
 
 import cats.syntax.all.*
+import lucuma.core.model.Observation
+import lucuma.core.util.Enumerated
 import observe.model.ClientId
+import observe.model.Observer
+import observe.model.enums.Instrument
 
 object ClientIDVar {
   def unapply(str: String): Option[ClientId] =
     Either.catchNonFatal(ClientId(java.util.UUID.fromString(str))).toOption
 }
+
+trait EnumeratedVar[A: Enumerated] {
+  def unapply(str: String): Option[A] =
+    Enumerated[A].all.find(a => Enumerated[A].fromTag(str).exists(_ === a))
```

Comment:
Why not just use `Enumerated[A].fromTag` ?

Reply:
uh